be/src/util/utf8.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "util/utf8.h"
+
+#include <unicode/ucasemap.h>
+
+#include <algorithm>
+#include <cctype>
+
+#include "common/logging.h"
+
+namespace starrocks {
+
+void utf8_tolower(const char* src, size_t src_len, std::string& dst) {
+    UErrorCode err_code = U_ZERO_ERROR;
+    UCaseMap* case_map = ucasemap_open("", U_FOLD_CASE_DEFAULT, &err_code);
```

Comment:
Is this UCaseMap real map like std::map?
the map should be readonly, so it can be loaded only once before using it.